apps/sel4test-tests/src/tests/scheduler.c

```diff
@@ -1697,3 +1697,65 @@ static int test_changing_affinity(struct env *env)
 }
 DEFINE_TEST(SCHED0022, "test changing a helper threads core", test_changing_affinity,
             (config_set(CONFIG_KERNEL_MCS) &&(CONFIG_MAX_NUM_NODES > 1)));
+
+static void yielding_thread(int id, volatile unsigned long *counters)
+{
+    counters[id] = 0;
+    while (1) {
+        counters[id]++;
+        seL4_Yield();
+    }
+}
+
+#define M   (200) // Minimum budget
+#define P   (2 * M)
+#define N   (US_IN_S / P)
+
+static int run_test_period_drift(env_t env, int num_threads)
+{
+    int error;
+    helper_thread_t t[num_threads];
+    volatile unsigned long counters[num_threads];
+
+    /* Create tasks with lower priority: */
+    for (int i = 0; i < num_threads; i++) {
+        create_helper_thread(env, &t[i]);
+        set_helper_priority(env, &t[i], env->priority - 1);
```

Comment:
maybe add an assert here that this does not cause a roll over.

Reply:
Good idea. `test_timeout_fault_in_server` does `OUR_PRIO - 2`, and `#define OUR_PRIO (env->priority)`. It seems priority is set to `seL4_MaxPrio - 1` in main.c.